nncf/common/tensor_statistics/statistics_serializer.py

```diff
@@ -8,112 +8,141 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import gzip
 import json
-import pickle
 import re
+from collections import defaultdict
 from pathlib import Path
-from typing import Any, Dict, Optional, Tuple, cast
+from typing import Any, Dict, List, Optional, TextIO, Tuple, cast
 
 import nncf
-from nncf.common.utils.os import fail_if_symlink
 from nncf.common.utils.os import safe_open
+from nncf.tensor import functions as fns
+from nncf.tensor.definitions import TensorBackend
+from nncf.tensor.tensor import Tensor
 
 METADATA_FILE = "statistics_metadata.json"
 
+# - Metadata is stored in a JSON file named "statistics_metadata.json".
```

Comment:
Could you mention why you need to use metadata if you use `safetensors` library with header inside?

Reply:
to store common information regarding caching: backend, subset_size. Can be extended in many ways

Reply:
ok